google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -432,13 +444,26 @@ private void publishAllWithoutInflightForKey(final String orderingKey) {
   }
 
   private ApiFuture<PublishResponse> publishCall(OutstandingBatch outstandingBatch) {
-    return publisherStub
-        .publishCallable()
-        .futureCall(
-            PublishRequest.newBuilder()
-                .setTopic(topicName)
-                .addAllMessages(outstandingBatch.getMessages())
-                .build());
+    if (enableCompression && outstandingBatch.batchSizeBytes >= compressionBytesThreshold) {
+      GrpcCallContext context = GrpcCallContext.createDefault();
+      context = context.withCallOptions(CallOptions.DEFAULT.withCompression("gzip"));
```

Comment:
Could we move the "gzip" to a class constant?

Not sure if this is out of scope, but would we ever want to use a different compression?

Reply:
made it constant. currently only gzip compression is supported. We don't have plan to add support for other compression.